protobuf/plugin/raftproxy/raftproxy.go

```diff
@@ -244,19 +228,33 @@ func (g *raftProxyGen) genClientServerStreamingMethod(s *descriptor.ServiceDescr
 func (g *raftProxyGen) genSimpleMethod(s *descriptor.ServiceDescriptorProto, m *descriptor.MethodDescriptorProto) {
 	g.gen.P(sigPrefix(s, m) + "ctx context.Context, r *" + getInputTypeName(m) + ") (*" + getOutputTypeName(m) + ", error) {")
 	g.gen.P(`
-	if p.cluster.IsLeader() {
-			return p.local.` + m.GetName() + `(ctx, r)
-	}
-	ctx, err := p.runCtxMods(ctx)
+	conn, err := p.connSelector.LeaderConn(ctx)
 	if err != nil {
+		if err == raftselector.ErrIsLeader {
+			return p.local.` + m.GetName() + `(ctx, r)
+		}
 		return nil, err
 	}
-	conn, err := p.connSelector.Conn()
+	ctx, err = p.runCtxMods(ctx)
 	if err != nil {
 		return nil, err
 	}`)
-	g.gen.P(connErrorHandle)
-	g.gen.P("return New" + s.GetName() + "Client(conn)." + m.GetName() + "(ctx, r)")
+	g.gen.P(`
+	resp, err := New` + s.GetName() + `Client(conn).` + m.GetName() + `(ctx, r)
+	if err != nil {
+		if !strings.Contains(err.Error(), "is closing") {
+			return resp, err
+		}
+		conn, err := p.pollNewLeaderConn(ctx, conn)
+		if err != nil {
+			if err == raftselector.ErrIsLeader {
+				return p.local.` + m.GetName() + `(ctx, r)
+			}
+			return nil, err
+		}
+		return New` + s.GetName() + `Client(conn).` + m.GetName() + `(ctx, r)
+	}`)
```

Comment:
I think it will complicate code a lot and every other piece apart from users of controlapi has retries.